BootloaderCommonPkg/Library/PagingLib/PagingMap.c

```diff
@@ -20,10 +22,67 @@
 **/
 UINT32
 GetPageShiftBits (
-  VOID
+  IN  UINT8   Level
 )
 {
-  return IsLongModeEnabled() ? 21 : 22;
+  if (Level == PDE_PG_LVL) {
+    return IsLongModeEnabled() ? 21 : 22;
+  } else {
+    return IsLongModeEnabled() ? 30 : 31;
+  }
+}
+
+EFI_STATUS
+AddPdpEntry (

```

Comment:
Add function description

Reply:
And the function name needs to be more specific
Maybe  MapAddressByHugePage () ?